src/access/AccessControl.cpp

```diff
@@ -23,92 +23,144 @@ namespace {
 using chip::FabricIndex;
 using namespace chip::Access;
 
-// Avoid GetAccessControl returning nullptr before SetAccessControl is called.
-class UnimplementedDataProvider : public AccessControlDataProvider
+AccessControl defaultAccessControl;
+AccessControl * globalAccessControl = &defaultAccessControl;
+
+static_assert((int(Privilege::kAdminister) & int(Privilege::kManage)) == 0);
+static_assert((int(Privilege::kAdminister) & int(Privilege::kOperate)) == 0);
+static_assert((int(Privilege::kAdminister) & int(Privilege::kView)) == 0);
+static_assert((int(Privilege::kAdminister) & int(Privilege::kProxyView)) == 0);
+static_assert((int(Privilege::kManage) & int(Privilege::kOperate)) == 0);
+static_assert((int(Privilege::kManage) & int(Privilege::kView)) == 0);
+static_assert((int(Privilege::kManage) & int(Privilege::kProxyView)) == 0);
+static_assert((int(Privilege::kOperate) & int(Privilege::kView)) == 0);
+static_assert((int(Privilege::kOperate) & int(Privilege::kProxyView)) == 0);
+static_assert((int(Privilege::kView) & int(Privilege::kProxyView)) == 0);
+
+int GetGrantedPrivileges(Privilege privilege)
 {
-    CHIP_ERROR Init() override { return CHIP_NO_ERROR; }
-
-    void Finish() override {}
-
-    EntryIterator * Entries() const override { return nullptr; }
-
-    EntryIterator * Entries(FabricIndex fabricIndex) const override { return nullptr; }
-};
-
-// Avoid GetAccessControl returning nullptr before SetAccessControl is called.
-UnimplementedDataProvider gUnimplementedDataProvider;
-AccessControl gUnimplementedAccessControl(gUnimplementedDataProvider);
-
-AccessControl * gAccessControl = &gUnimplementedAccessControl;
+    switch (privilege)
+    {
+        case Privilege::kView:
+            return int(Privilege::kView);
+        case Privilege::kProxyView:
+            return int(Privilege::kProxyView) | int(Privilege::kView);
+        case Privilege::kOperate:
+            return int(Privilege::kOperate) | int(Privilege::kView);
+        case Privilege::kManage:
+            return int(Privilege::kManage) | int(Privilege::kOperate) | int(Privilege::kView);
+        case Privilege::kAdminister:
+            return int(Privilege::kAdminister) | int(Privilege::kManage) | int(Privilege::kOperate) | int(Privilege::kView) | int(Privilege::kProxyView);
+    }
+    return 0;
+}
 
 } // namespace
 
 namespace chip {
 namespace Access {
 
+AccessControl::Entry::Delegate AccessControl::Entry::mDefaultDelegate;
+AccessControl::EntryIterator::Delegate AccessControl::EntryIterator::mDefaultDelegate;
+AccessControl::Delegate AccessControl::mDefaultDelegate;
```

Comment:
Can we avoid having these global, and make them a member of AccessControl class?

Reply:
I'm not sure what you mean, they aren't global, they are private static members. Do you prefer another naming convention for static members?

Reply:
They are static so they can be set in the constructor, and reset in a function. I do use both of those capabilities. Therefore I prefer to keep them here. I can rename them to use an 's' prefix if that is preferable.